pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -84,6 +89,29 @@ func (c *EtcdEndpointsController) sync(ctx context.Context, syncCtx factory.Sync
 }
 
 func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder events.Recorder) error {
+	// This resource should have been created at installation and should never
+	// be deleted. Treat the inability to access it as fatal to smoke out any
+	// problems which could indicate the resource is gone.
+	existing, err := c.configmapLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
+	if err != nil {
+		return fmt.Errorf("couldn't get required configmap %s/%s: %w", operatorclient.TargetNamespace, "etcd-endpoints", err)
+	}
+	_, existingHasBootstrapIP := existing.Annotations[etcdcli.BootstrapIPAnnotationKey]
+
+	// If the bootstrap IP annotation appears to have been prematurely removed, generate a warning.
+	// This could be a transient condition depending on the state of caches at the time of sync,
+	// so a warning seems sufficient. This check may be overly paranoid, but a significant amount
+	// of time has been spent in the past tracing problems back to unexpected deletions and
+	// modifications of the endpoints structure (previously a literal endpoints resource).
+	isSafe, err := c.isSafeToRemoveBootstrapIP()
+	if err != nil {
+		return fmt.Errorf("failed to determine whether it's safe to remove the boostrap IP annotation: %w", err)
+	}
+	if !existingHasBootstrapIP && !isSafe {
+		recorder.Warningf("EtcdEndpointsMissingBootstrapIP", "the bootstrap IP annotation has been removed from the %s/%s "+
```

Comment:
doesn't look very helpful. it won't be safe to remove generally because the count is wrong